### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by TotalDwarf03**
Changes include:
- Modifying the code to use a python package to interact with the GH API instead of stored local version
- Changing any background variables to come from the environment instead of being written in code
- Add MkDocs support
- Bug: Removed reference to old /login in error.html
- Feat: Can now search in /manage_repositories by multiple columns at once
- Feat: Added feature selection to enable/disable /insert_test_data

### CODE REVIEW

**Review by WojciechP-ONS — APPROVED** [review]
Everything looks good to me, happy to approve this PR

File: repoarchivetool/data_retrieval.py [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+
+        Returns:
+            str: An error message.
+            or
+            list: A list of dictionaries containing information about the repositories collected from 
+            the Github API.
+    """
+
+    
+    def archive_flag(repo_url: str, comp_date: date) -> bool | str:
```
Comment (gibbardsteve) [inline repoarchivetool/data_retrieval.py:39]: Does this need to be a nested function? Can we move it out from get_organisation_repos as I find it harder to read the intent of that function.

File: repoarchivetool/static/js/main.js [hunk 2]
```diff
@@ -19,100 +21,113 @@ function deleteNoResults(table){
...
+            insertNoResults(table);
+        }
+        else {
+            deleteNoResults(table);
+        }
     }
+
 }
 
+// recentlyArchived.html
```
Comment (gibbardsteve) [inline repoarchivetool/static/js/main.js:129]: ?

**Review by gibbardsteve — COMMENTED** [review]
A couple of minor comments to tidy.

File: pyproject.toml [hunk 3]
```diff
@@ -13,6 +13,7 @@ flask = "^3.0.2"
 python-dateutil = "^2.9.0.post0"
 jwt = "^1.3.1"
 boto3 = "^1.34.118"
+github-api-toolkit = {git = "https://github.com/ONS-Innovation/github-api-package.git"}
```
[the remark this answers is not in the document]
Comment (TotalDwarf03) [reply pyproject.toml:16]: Ticket created - Will amend next sprint

**Pull request merged**